shell/src/main/java/org/apache/accumulo/shell/commands/FateCommand.java

```diff
@@ -287,6 +296,34 @@ protected boolean cancelSubmittedTxs(final Shell shellState, String[] args)
     return true;
   }
 
+  private static boolean cancelFateOperation(ClientContext context, long txid,
+      final Shell shellState) throws AccumuloException, AccumuloSecurityException {
+    while (true) {
+      FateService.Client client = null;
+      try {
+        client = ThriftClientTypes.FATE.getManagerConnectionWithRetry(context);
+        return client.cancelFateOperation(TraceUtil.traceInfo(), context.rpcCreds(), txid);
+      } catch (TTransportException tte) {
+        shellState.getWriter()
+            .println("ManagerClient request failed, retrying. Cause: " + tte.getMessage());
+        sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
```

Comment:
I think we need to abstract things a bit further, if we're leaking these Thrift types into the shell. That implies we're not doing a great job of limiting ourselves to public API types.

I actually don't think we need to do as much work as we're doing here to retry any of this or to convert all the exceptions here. The shell is for humans... we can just dump any stack trace and the user can press the up arrow and try again if they need to.

Reply:
Suggestion implemented